### DISCUSSION TITLE
How can fuse server exits gracefully?

### DISCUSSION THREAD

**Discussion by yangyun50** [libfuse/libfuse#986] [Q&A]
I want to know which is the right way to exit fuse server gracefully. And I think it may be useful for beginners to add these information to the Doc (e.g., Readme.md).
1. signal (e.g., ctrl+c)
In libfuse, sending signal to fuse server can cause the process to exit since libfuse set the customized signal handler for SIGINT, SIGHUP, and SIGTERM. 
But what about SIGKILL? If I use SIGKILL, then the mountpoint may be not umounted？This may lead to the fuse server exit ungracefully.
2. umount / fusermount
I am not familar with fusermount. So can we just use umount to make the fuse server exit?  
3. echo 1 > /sys/fs/fuse/connections
I know this is emergency exit way export by the kernel.

**Comment by trapexit** [comment]
It is 2. Same as a normal filesystem. You should umount or fusermount -u the filesystem.
